webviz_subsurface/plugins/_group_tree/group_tree.py

```diff
@@ -0,0 +1,215 @@
+        )
+        return functions
+
+    @property
+    def tour_steps(self) -> List[dict]:
+        return [
+            {
+                "id": self.uuid("layout"),
+                "content": "Dashboard vizualizing Eclipse network tree.",
+            },
+            {
+                "id": self.uuid("selections_layout"),
+                "content": "Menu for selecting ensemble and other options.",
+            },
+            {
+                "id": self.uuid("filters_layout"),
+                "content": "Menu for filtering options.",
+            },
+            {
+                "id": self.uuid("grouptree_wrapper"),
+                "content": "Vizualisation of network tree.",
+            },
+        ]
+
+    @property
+    def layout(self) -> html.Div:
+        return html.Div(
+            children=[
+                # clientside_stores(get_uuid=self.uuid),
+                main_view(get_uuid=self.uuid, ensembles=self.ensembles),
+            ],
+        )
+
+    def set_callbacks(self, app: dash.Dash) -> None:
+        controllers(app=app, get_uuid=self.uuid, grouptreedata=self.grouptreedata)
+
+
+@webvizstore
+def read_gruptree_files(ens_paths: Dict[str, str], gruptree_file: str) -> pd.DataFrame:
+    """Searches for gruptree files on the scratch disk. These
+    files can be exported in the FMU workflow using the ECL2CSV
+    forward job with subcommand gruptree.
+    """
+    df = pd.DataFrame()
+    for ens_name, ens_path in ens_paths.items():
+        df_ens = read_ensemble_gruptree(ens_name, ens_path, gruptree_file)
+        df_ens["ENSEMBLE"] = ens_name
+        df = pd.concat([df, df_ens])
+    return df.where(pd.notnull(df), None)
+
+
+def read_ensemble_gruptree(
+    ens_name: str, ens_path: str, gruptree_file: str
+) -> pd.DataFrame:
+    """Reads the gruptree file for and ensemble.
+
+    If BRANPROP is found in the KEYWORD column, then GRUPTREE rows
+    are filtered out.
+
+    If the trees are equal in every realization, only one realization is kept.
+    """
+    ens = ScratchEnsemble("ens", ens_path)
+    df_files = ens.find_files(gruptree_file)
+
+    if df_files.empty:
+        raise ValueError(f"No gruptree file available for ensemble: {ens_name}")
+
+    # Load all gruptree dataframes and check if they are equal
+    compare_columns = ["DATE", "CHILD", "KEYWORD", "PARENT"]
+    df_prev = pd.DataFrame()
+    df_all = pd.DataFrame()
+    gruptrees_are_equal = True
+    for i, row in df_files.iterrows():
+        df_real = pd.read_csv(row["FULLPATH"])
+
+        if "BRANPROP" in df_real["KEYWORD"].unique():
+            df_real = df_real[df_real["KEYWORD"] != "GRUPTREE"]
+        if (
+            i > 0
+            and gruptrees_are_equal
+            and not df_real[compare_columns].equals(df_prev)
+        ):
+            gruptrees_are_equal = False
+        else:
+            df_prev = df_real[compare_columns].copy()
+
+        df_real["REAL"] = row["REAL"]
+        df_all = pd.concat([df_all, df_real])
```

Comment:
Would guess that it is faster to temporarily store a list of all df's for realizations and then concat all at once, rather than concatenating a realization df with a growing "total df" for each realization (probably not massive data amounts here, so could be that it is negligible, but percentage wise for this step I would think it is significant).